core/kazoo_ast/src/cb_api_endpoints.erl

```diff
@@ -0,0 +1,731 @@
+                end
+               ,Acc
+               ,Paths
+               ).
+
+format_pc_callback({[], []}, Acc, _Module, _ModuleName, _Callback) ->
+    Acc;
+format_pc_callback({Path, []}, Acc, _Module, ModuleName, Callback) ->
+    PathName = path_name(Path, ModuleName),
+    kz_json:set_value([PathName, kz_util:to_binary(Callback)], <<"not supported">>, Acc);
+format_pc_callback({Path, Vs}, Acc, Module, ModuleName, Callback) ->
+    PathName = path_name(Path, ModuleName),
+    kz_json:set_values(
+      props:filter_undefined(
+        [{[PathName, kz_util:to_binary(Callback)]
+         ,[kz_util:to_lower_binary(V) || V <- Vs]
+         }
+        ,maybe_include_schema(PathName, Module)
+        ]
+       )
+                      ,Acc
+       ).
+
+maybe_include_schema(PathName, Module) ->
+    M = base_module_name(Module),
+    case filelib:is_file(?SCHEMAS_PATH(M)) of
+        'true' ->
+            {[PathName, <<"schema">>], base_module_name(Module)};
+        'false' ->
+            {'undefined', 'undefined'}
+    end.
+
+format_path_tokens(<<"/">>) -> [];
+format_path_tokens(<<_/binary>> = Token) ->
+    [format_path_token(Token)];
+format_path_tokens(Tokens) ->
+    [format_path_token(Token) || Token <- Tokens, Token =/= <<"/">>].
+
+format_path_token(<<"_">>) -> <<"{ID}">>;
+format_path_token(<<"AccountId">>) -> <<"{ACCOUNT_ID}">>;
+format_path_token(<<"_UUID">>) -> <<"{UUID}">>;
+format_path_token(<<"_", Rest/binary>>) ->
+    VarName = kz_util:to_upper_binary(Rest),
+    case binary:split(VarName, <<"ID">>) of
+        [Thing, <<>>] -> <<"{", Thing/binary, "_ID}">>;
+        _ -> <<"{", (VarName)/binary, "}">>
+    end;
+format_path_token(Token) -> Token.
+
+base_module_name(Module) when is_atom(Module) ->
+    base_module_name(kz_util:to_binary(Module));
+base_module_name(<<_/binary>>=Module) ->
+    {'match', [Name|_]} = re:run(Module
+                                ,<<"^cb_([a-z_]+?)(?:_v([0-9]))?$">>
+                                ,[{'capture', 'all_but_first', 'binary'}]
+                                ),
+    Name.
+
+module_version(Module) when is_atom(Module) ->
+    module_version(kz_util:to_binary(Module));
+module_version(<<_/binary>> = Module) ->
+    case re:run(Module
+               ,<<"^cb_([a-z_]+?)(?:_(v[0-9]))?$">>
+               ,[{'capture', 'all_but_first', 'binary'}]
+               )
+    of
+        {'match', [_Name, Version]} -> Version;
+        {'match', [_Name]} -> ?CURRENT_VERSION
+    end.
+
+path_name(Path, ModuleName) ->
+    kz_util:join_binary([<<>>, ModuleName | format_path_tokens(Path)], <<"/">>).
+
+path_name(Module) when is_atom(Module) ->
+    path_name(kz_util:to_binary(Module));
+path_name(<<_/binary>>=Module) ->
+    case re:run(Module
+               ,<<"^cb_([a-z_]+?)(?:_(v[0-9]))?$">>
+               ,[{'capture', 'all_but_first', 'binary'}]
+               )
+    of
+        {'match', [<<"about">>]} -> <<?CURRENT_VERSION/binary, "/about">>;
```

Comment:
What is this doing here? It was removed earlier this week. Did you really `git mv` the file?